scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/dao/GenericDaoImpl.java

```diff
@@ -25,33 +27,45 @@ public GenericDaoImpl(Class<T> entityClass) {
 
 	@Override
 	public T create(T entity) {
-		entityManager.persist(entity);
-		entityManager.flush();
+		try{
+			entityManager.persist(entity);
+			entityManager.flush();
+		} catch(Exception exception){
+			throw new DatabaseValidationException("Create Failed", exception);
+		}
 		return entity;
 	}
 
 	@Override
 	public T update(T entity) {
 		try {
 			return entityManager.merge(entity);
-		} catch (Exception ex) {
-			return null;
+		} catch(Exception exception){
+			throw new DatabaseValidationException("Update Failed", exception);
 		}
 	}
 
 	@Override
 	public Boolean delete(T entity) {
+		boolean deleted = false;
 		try {
 			entityManager.remove(entity);
-		} catch (Exception ex) {
-			return false;
+			deleted = true;
+		} catch(Exception exception){
+			throw new DatabaseValidationException("Delete failed", exception);
 		}
-		return true;
+		return deleted;
```

Comment:
Yes thats incorrect. Added the exception handling but did not clean the code, same for the following comment. Fixed other methods as well.